lib/rules/no-template-children.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { docUrl } = require('../util/doc-url');
+
+module.exports = {
+    meta: {
+        docs: {
+            description: 'prevent accessing the immediate children of this.template',
+            category: 'LWC',
+            recommended: true,
+            url: docUrl('no-template-children'),
+        },
+
+        schema: [],
+    },
+
+    create(context) {
+        const disallowedProperties = ['firstElementChild', 'firstChild', 'children', 'childNodes'];
```

Comment:
what do you think about making this list a configuration for the rule?

Reply:
I dunno, I can't think of a case where someone would want to add or remove things from this list? It seems to me that anything that directly access the child is unsafe. In fact we could probably add `lastChild`/`lastElementChild` as well.

Reply:
Added `lastChild`/`lastElementChild`. I think it's simplest if we just disallow any reference to any of the children.

Also in the future we should add `ref` so that people don't need `querySelector`!